marketplace-builder/src/builder.rs

```diff
@@ -332,54 +360,79 @@ mod test {
+        let solver_base_url = mock_solver.solver_url.clone();
+        let client = connect_to_solver(solver_base_url.clone());
+
+        // Create a list of signature keys for rollup registration data
+        let mut signature_keys = Vec::new();
+
+        for _ in 0..10 {
+            let private_key =
+                <BLSPubKey as SignatureKey>::PrivateKey::generate(&mut rand::thread_rng());
+            signature_keys.push(BLSPubKey::from_private(&private_key))
+        }
 
-        let config = TestNetworkConfigBuilder::default()
-            .api_config(
-                Options::with_port(query_port)
-                    .submit(Default::default())
-                    .query_fs(
-                        Default::default(),
-                        persistence::fs::Options::new(tmpdir.path().to_owned()),
-                    )
-                    .hotshot_events(HotshotEvents {
-                        events_service_port: event_port,
-                    }),
-            )
-            .network_config(network_config)
-            .build();
-        let _network = TestNetwork::new(config, MockSequencerVersions::new()).await;
+        let private_key =
+            <BLSPubKey as SignatureKey>::PrivateKey::generate(&mut rand::thread_rng());
+        let signature_key = BLSPubKey::from_private(&private_key);
+
+        signature_keys.push(signature_key);
+
+        // Initialize a rollup registration with namespace id = 10
+        let reg_ns_1_body = RollupRegistrationBody {
+            namespace_id: REGISTERED_NAMESPACE.into(),
+            reserve_url: Some(Url::from_str("http://localhost").unwrap()),
+            reserve_price: 200.into(),
+            active: true,
+            signature_keys: signature_keys.clone(),
+            text: "test".to_string(),
+            signature_key,
+        };
 
-        // Start the builder
-        let init = BuilderConfig::init(
-            true,
-            FeeAccount::test_key_pair(),
-            ViewNumber::genesis(),
-            NonZeroUsize::new(1024).unwrap(),
-            NonZeroUsize::new(1024).unwrap(),
-            NodeState::default(),
-            ValidatedState::default(),
-            event_service_url.clone(),
-            builder_api_url.clone(),
-            Duration::from_secs(2),
-            5,
-            Duration::from_secs(2),
-            FeeAmount::from(10),
-            Some(BidConfig {
-                namespaces: vec![NamespaceId::from(10u32)],
-                amount: FeeAmount::from(10),
-            }),
-            format!("http://localhost:{}", 3000).parse().unwrap(),
-        );
-        let _builder_config = init.await;
+        // Sign the registration body
+        let reg_signature = <SeqTypes as NodeType>::SignatureKey::sign(
+            &private_key,
+            reg_ns_1_body.commit().as_ref(),
+        )
+        .expect("failed to sign");
+
+        let reg_ns_1 = RollupRegistration {
+            body: reg_ns_1_body.clone(),
```

Comment:
Small nit: Does this clone need to be here?  It seems like reg_ns_1 can take ownership of reg_ns_body?